File: services/order-service/src/main/java/io/minishop/order/web/dto/OrderResponse.java
package io.minishop.order.web.dto;

import io.minishop.order.domain.Order;
import io.minishop.order.domain.OrderStatus;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

public record OrderResponse(
        Long id,
        Long userId,
        OrderStatus status,
        BigDecimal totalAmount,
        List<OrderItemResponse> items,
        Instant createdAt,
        Instant updatedAt
) {
    public static OrderResponse from(Order order) {
        return new OrderResponse(
                order.getId(),
                order.getUserId(),
                order.getStatus(),
                order.getTotalAmount(),
                order.getItems().stream().map(OrderItemResponse::from).toList(),
                order.getCreatedAt(),
                order.getUpdatedAt()
        );
    }
}

File: services/order-service/src/main/kotlin/io/minishop/order/web/dto/OrderResponse.kt
package io.minishop.order.web.dto

import io.minishop.order.domain.Order
import io.minishop.order.domain.OrderStatus
import java.math.BigDecimal
import java.time.Instant

@JvmRecord
data class OrderResponse(
    val id: Long?,
    val userId: Long?,
    val status: OrderStatus?,
    val totalAmount: BigDecimal?,
    val items: List<OrderItemResponse>,
    val createdAt: Instant?,
    val updatedAt: Instant?,
) {
    companion object {
        @JvmStatic
        fun from(order: Order): OrderResponse = OrderResponse(
            order.id,
            order.userId,
            order.status,
            order.totalAmount,
            order.items.map { OrderItemResponse.from(it) },
            order.createdAt,
            order.updatedAt,
        )
    }
}
